src/ComputeClosestToZeroTemperature/index.ts

```diff
@@ -0,0 +1,32 @@
+export class ComputeClosestToZeroTemperature {
+  private convertNegativeToPositiveValue(value: number): number {
+    return value < 0 ? value * -1 : value;
+  }
+  static check(temperatures: number[]): number {
+    if (!temperatures || temperatures.length < 1) return 0;
+    if (temperatures.length === 1) return temperatures[0];
+
+    // if array has more than one element
+    let closestTarget =
+      new ComputeClosestToZeroTemperature().convertNegativeToPositiveValue(
+        temperatures[0]
+      );
+    let targetIndex = -1;
+
+    for (let index = 0; index < temperatures.length; index++) {
+      const currentPositiveTemp =
+        new ComputeClosestToZeroTemperature().convertNegativeToPositiveValue(
+          temperatures[index]
+        );
+      if (currentPositiveTemp < closestTarget) {
+        closestTarget = currentPositiveTemp;
+        targetIndex = index;
+      } else if (currentPositiveTemp === closestTarget) {
+        closestTarget = currentPositiveTemp;
+      }
+    }
+
+    if (targetIndex !== -1) closestTarget = temperatures[targetIndex];
+    return closestTarget;
```

Comment:
I think much simpler would to do would be this and it only take one line

```suggestion
return Math.min(temperatures.map(Math.abs))
```

Reply:
`static get(temperatures: number[]): number {
    if (!temperatures?.length) return 0;

    return temperatures.reduce((closest: number, current: number) => {
      let closestAbs = Math.abs(closest);
      let currentAbs = Math.abs(current);
      if (closestAbs === currentAbs) {
        closest = current > 0 ? current : currentAbs
      }
      if (currentAbs < closestAbs) {
        closest = current;
      }
      return closest
    })
  }`
  You are right. It is better to use .reduce() method.
  I have tuned the code as above.

Reply:
```suggestion
  export class ComputeClosestToZeroTemperature {
  static get(temperatures: number[]): number {
    if (!temperatures?.length) return 0;
    return temperatures.reduce((closestValue: number, currentValue: number) => {
      const positiveClosestValue = Math.abs(closestValue)
      if (Math.abs(currentValue) < positiveClosestValue || currentValue === positiveClosestValue) return currentValue;
      return closestValue
    })
  }
}
```